.goreleaser-build.yml

```diff
@@ -5,3 +5,5 @@ builds:
     main: cmd/confluent/main.go
     flags:
       - -tags={{.Env.TAGS}}
+    ldflags:
+      - "-X local.GetLocalArch={{.Env.LOCAL_ARCH}}"
```

Comment:
no. It'll always be the goarch of your machine so amd64. i have to set the var in the package

Reply:
Why are we trying to run an arm64 image on an amd64 machine?

Reply:
that's the bug. if it's `amd64` when being built, since cli is a static binary, it'll also be amd64 when user runs the binary on arm64, then cli will try to create a amd64 container on their arm64 machine. we need this value to be the arch of the platform we're building for, when releasing cli

Reply:
We use `runtime.GOARCH` in a few other places, and it's always arm64 when we're running on arm64...

Reply:
I agree with Brian. Aside from `goreleaser-build.yml`, the builds in the rest of our goreleaser files specify the `goos` and `goarch` values, and those should be the ones recorded in the runtime package's constants.

The update logic also depends on `runtime.GOARCH`, so if that wasn't arm64 in the arm64 binaries then updates would probably break.

Reply:
Random guessing: is it possible to run the amd64 binary on arm64 macs? That might cause the image mismatch with `wanted linux/amd64, actual: linux/arm64`.